ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/editor/filetype/ExtensionFileTypeIdentifier.java

```diff
@@ -10,21 +10,29 @@
  */
 package org.eclipse.che.ide.api.editor.filetype;
 
+import com.google.common.collect.ImmutableList;
 import java.util.ArrayList;
-import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import javax.inject.Inject;
+import javax.inject.Singleton;
 import org.eclipse.che.ide.api.resources.VirtualFile;
 import org.eclipse.che.ide.util.loging.Log;
 
+@Singleton
 public class ExtensionFileTypeIdentifier implements FileTypeIdentifier {
 
   /** The known extensions registry. */
   Map<String, List<String>> mappings = new HashMap<>();
 
-  public ExtensionFileTypeIdentifier() {
```

Comment:
It would have been a much simpler fix to simply register the mime types for hover here instead of refactoring stuff.

Reply:
As I remember we have already discussed this in another pull request, in general my attitude stays the same - my preference is to do it the most appropriate and simple way, not just simple way. If you experience difficulties in understanding I would recommend you to become more familiar with such concepts as [decomposition](https://en.wikipedia.org/wiki/Decomposition_(computer_science)), [code refactoring](https://en.wikipedia.org/wiki/Code_refactoring) and [single responsibility principle](https://en.wikipedia.org/wiki/Single_responsibility_principle). In our case there are several quite different components and their corresponding responsibilities: file type identification, default mime types registration, hover configuration initialization. Please just ping me if you need any other assistance.